basic_bootloader/src/bootloader/gas_helpers.rs

```diff
@@ -39,9 +39,25 @@ pub fn get_resources_for_tx<S: EthereumLikeTypes>(
                 ),
             ))?;
 
-    // EVM tester requires high native limits
+    // EVM tester requires high native limits, so for it we never hold off resources.
+    // But for the real world, we bound the available resources.
+
+    let withheld_resources = S::Resources::from_ergs(Ergs(0));
+
     #[cfg(not(feature = "resources_for_tester"))]
-    let native_limit = native_limit.min(MAX_NATIVE_COMPUTATIONAL);
+    let (native_limit, withheld_resources) = if native_limit <= MAX_NATIVE_COMPUTATIONAL {
+        (native_limit, S::Resources::from_ergs(Ergs(0)))
+    } else {
+        let withheld =
+            <<S as zk_ee::system::SystemTypes>::Resources as Resources>::Native::from_computational(
+                native_limit - MAX_NATIVE_COMPUTATIONAL,
+            );
+
+        (
+            MAX_NATIVE_COMPUTATIONAL,
+            S::Resources::from_native(withheld),
```

Comment:
Can you also call set_as_limit for withheld? 
for Decreasing native it doesn’t matter, but it’s useful for Increasing (which are used for benchmarking). Then, in the reclaim_method, we should also set the limit to the sum of self and to_reclaim